File: jar-shell/src/main/java/com/beancounter/shell/ingest/RowAdapter.java
package com.beancounter.shell.ingest;

import com.beancounter.common.input.TrnInput;
import com.beancounter.common.model.Portfolio;
import java.util.List;

public interface RowAdapter {
  TrnInput transform(Portfolio portfolio,
                                 List<String> values,
                                 String provider);

}

File: jar-common/src/main/kotlin/com/beancounter/client/ingest/RowAdapter.kt
package com.beancounter.client.ingest

import com.beancounter.common.input.TrnInput
import com.beancounter.common.input.TrustedTrnImportRequest

/**
 * Transform the TrustedTrnImportRequest payload into a writable TrnInput
 */
interface RowAdapter {
    fun transform(trustedTrnImportRequest: TrustedTrnImportRequest): TrnInput
}